internal/provider/resource_resource_class.go

```diff
@@ -142,6 +142,12 @@ func (r *ResourceResourceClass) Read(ctx context.Context, req resource.ReadReque
 		return
 	}
 
+	if httpResp.StatusCode() == 404 {
+		resp.Diagnostics.AddWarning("Resource class not found", fmt.Sprintf("The resource class (%s) was deleted outside Terraform", data.ID.ValueString()))
```

Comment:
nitpick: I wonder if we should rather just say "The resource class (%s) does not exist"?

How do we know the resource actually ever existed? espescially if someone is using terraform state importing to add a resource they created by hand?

Reply:
> How do we know the resource actually ever existed?

The case here is that the resource is in the TF state but we can't read it because we don't find it anymore.
I think it would be helpful to see this message in that case, no?